docs/blog/index.mdx

```diff
@@ -1,3 +1,12 @@
+{/**
+  * @import {Item} from '../_component/sort.js'
+  */}
+
+{/**
+  * @typedef Props
+  * @property {Item} navigationTree
+  */}
```

Comment:
not 100% on this style? I’d lean to:

```
{
  /**
   * @import {Item} from '../_component/sort.js'
   */

  /**
   * @typedef Props
   * @property {Item} navigationTree
   */
}
```

Reply:
Hmm, I feel quite strongly that entering/exiting JS each time is ugly.

And that if you’d put something in there (like a jsx element), that it would be indented 🤔

I’m fine to different typedefs in different comments.

Reply:
> Hmm, I feel quite strongly that entering/exiting JS each time is ugly.

I don’t really mind either style for typedefs. For function docs I would find it weird to combine though, but that’s not applicable in this particular case.

Another thing is that commenting is based on `{/*` and `*/}` delimiters. Though I don’t think it makes sense to uncomment jsdoc comments.

> And that if you’d put something in there (like a jsx element), that it would be indented 🤔

That makes perfect sense. This also reminds me that it’s possible to have multiple comments in one a flow expression, but just one JS expression. JSDoc has meaning, which makes me lean towards preferring one doc per flow expression again. But having the types grouped using one flow expression is also kind of nice.

---

Anyway, these are all just nitty details. I’ll adjust them to use one MDX flow expression.

Reply:
> For function docs I would find it weird to combine though, but that’s not applicable in this particular case.

Can you clarify what you dislike/like with examples?


> Though I don’t think it makes sense to uncomment jsdoc comments.

Agreed. These things are more like type annotations, which *currently* use special comments, because that’s what’s possible now.

> This also reminds me that it’s possible to have multiple comments in one a flow expression, but just one JS expression. JSDoc has meaning, which makes me lean towards preferring one doc per flow expression again. But having the types grouped using one flow expression is also kind of nice.

Right, right, I was thinking about that at first too, but then there’s a difference: these are allowed to be empty! And comments are deemed empty here too. 0 or more empty is still empty. So, zero-or-more empty, then 0-1 expression.

To clarify, some examples of what I think looks good.

Regular comment:

```mdx
# hi

{/* just a smol comment */}

hello
```

More complex expression:

```mdx
# hi

{
  (function () {
    return 1
  })()
}

hello
```

One JSDoc:

```mdx
{
  /**
   * @import {foo} from 'bar'
   */
}

# hi

hello
```

Multi JSDoc:

```mdx
{
  /**
   * @import {foo} from 'bar'
   */

  /**
   * @typedef Options
   * @property {number} something
   */
}

# hi

hello
```

Multi JSDoc and expression:

```mdx
# hi

{
  /**
   * @import {foo} from 'bar'
   */

  /**
   * @typedef Options
   * @property {number} something
   */

  (function () {
    return 1
  })()
}

hello
```

Also, was just thinking about how [do-expressions](https://github.com/tc39/proposal-do-expressions) will be nice!

```mdx
{
  do {
    if (loggedIn) {
      <LogoutButton />
    } else {
      <LoginButton />
    }
  }
}
```

Reply:
> Can you clarify what you dislike/like with examples?

If there are multiple exports with JSDoc annotations, It feels weird to me to merge the mdx flow expression the typedefs with that of the JSDoc of the first export.

```mdx
{
  /**
   * @typedef Props
   * @property {string} whatever
   */

  /**
   * @param {string} arg
   * @returns undefined
   */
}
export function someExport(arg) {}

{
  /**
   * @param {string} arg
   * @returns undefined
   */
}
export function anotherExport(arg) {}
```

> Also, was just thinking about how [do-expressions](https://github.com/tc39/proposal-do-expressions) will be nice!

Yes, that would be so nice!

Reply:
ok, I can see this, for this first example. But that feels like an edge case to me, induced because we allow `export` statements directly in markdown 🤔

I’m fine with breaking this like:

```mdx
{
  /**
   * @import {foo} from 'bar'
   */

  /**
   * @typedef Options
   * @property {number} something
   */
}

{
  /**
   * @param {string} arg
   * @returns {undefined}
   */
}
export function someExport(arg) {}
```